File: 2554-maximum-number-of-integers-to-choose-from-a-range-i/2554-maximum-number-of-integers-to-choose-from-a-range-i.cpp
class Solution {
public:
    int maxCount(vector<int>& banned, int n, int maxSum) {
        unordered_set<int> bannedSet(banned.begin(), banned.end());
        int count = 0;
        int sum = 0;
        for (int i = 1; i <= n; i++) {
            if (bannedSet.find(i) != bannedSet.end()) {
                continue;
            }
            if (sum + i > maxSum) {
                break;
            }
            sum += i;
            count++;
        }
        return count;
    }
};

File: 2554-maximum-number-of-integers-to-choose-from-a-range-i/2554-maximum-number-of-integers-to-choose-from-a-range-i.py
class Solution:
    def maxCount(self, banned: List[int], n: int, maxSum: int) -> int:
        banned_set = set(banned)
        count = 0
        total_sum = 0

        for i in range(1, n + 1):
            if i in banned_set:
                continue
            if total_sum + i > maxSum:
                break
            total_sum += i
            count += 1

        return count